src/regexp_object.cpp

```diff
@@ -403,7 +403,8 @@ void RegexpObject::initialize_internal(Env *env, const StringObject *pattern, in
     options &= ~RegexOpts::NoEncoding;
 
     UChar *pat = (UChar *)tweaked_pattern.c_str();
-    OnigEncoding enc = fixed_encoding && !no_encoding ? ruby_encoding_to_onig_encoding(pattern->encoding()) : ONIG_ENCODING_ASCII;
+    const auto enc = encoding ? ruby_encoding_to_onig_encoding(encoding) : fixed_encoding && !no_encoding ? ruby_encoding_to_onig_encoding(pattern->encoding())
+                                                                                                          : ONIG_ENCODING_ASCII;
```

Comment:
Holy nested ternary batman! 😆

Reply:
Yeah, it was more readable than this:
```c++
const auto enc = encoding ?
    ruby_encoding_to_onig_encoding(encoding) :
    fixed_encoding && !no_encoding ? ruby_encoding_to_onig_encoding(pattern->encoding()) :
    ONIG_ENCODING_ASCII;
```
But clang-format disagreed with me on that one.

On the bright side: there are a few other options related to encoding, so we might have to refactor this soon.